### PR TITLE
[ko] Update traslation

### PR DESCRIPTION
**Pull request by marocchino**
#676, #679, #680 
@Dalgona, @malkoG, @devleoper, @riseshia review please.

### CODE REVIEW

File: ko/lessons/basics/testing.md [hunk 1]
```diff
@@ -102,6 +142,8 @@ end
 
 ## 모의 객체 만들기
 
-Elixir에서 모의 객체(mock)를 만들어서 테스트를 진행하는 가장 깔끔한 방법은 "안 하는 것"입니다. 다른 언어에서 테스트를 다루어 보셨다면 지금쯤 모의 객체를 다룰 때가 되었다고 본능적으로 느끼셨겠지만, Elixir 커뮤니티에서 사용하지 말자는 의견이 크기도 하고, 사용하지 않을 만한 멋진 이유도 있기 때문이에요. 좋은 설계 원칙을 지켜 코딩을 한다면 그 결과물도 각 부분으로 나누어 테스트하기 쉬워질 테니까요.
+Elixir에서 모의 객체(mock)를 만들어서 테스트를 진행하는 가장 깔끔한 방법은 "안 하는 것"입니다. 다른 언어에서 테스트를 다루어 보셨다면 지금쯤 모의 객체를 다룰 때가 되었다고 본능적으로 느끼셨겠지만, Elixir 커뮤니티에서 사용하지 말자는 의견이 크기도 하고, 사용하지 않을 만한 멋진 이유도 있기 때문이에요.
+
+더 긴 논의가 보고 싶으시면, 이 [훌륭한 글](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/)이 있습니다. 요약하면, 테스트를 위해 의존성을 모의 객체로 만드는 것 보다, 명시적으로 인터페이스(비헤이비어)를 선언하는 것이 애플리케이션 외부의 코드에서 사용할 때나 테스트를 위한 클라이언트 코드에서의 모의 객체 구현에 많은 이점이 있다는 것입니다.
 
-모의 객체에 대한 끓어오르는 열망을 억누르도록 하세요.
+애플리케이션 코드에서 구현을 바꾸려면, 올바른 방법은 모듈을 인자로 넘기고 기본 값을 사용하는 것입니다. 이렇게 할 수 없으면, 내장된 설정 메카니즘을 사용하세요. 모의 객체를 구현하기위해 특별한 모의 객체 라이브러리를 사용할 필요가 없습니다. 그냥 비헤이비어와 콜백을 쓰세요.
```
Comment (riseshia) [inline ko/lessons/basics/testing.md]: 구현하기위해 -> 구현하기 위해

**Comment by riseshia** [comment]
I'm done XD

File: ko/lessons/basics/documentation.md [hunk 2]
```diff
@@ -124,6 +124,8 @@ iex>
 
 문서 내에서 마크업을 어떻게 이용하고, 터미널이 이를 어떻게 렌더링하는지 주목하세요. 더욱 세련되어지거나 Elixir 생태계에 멋진 기능들이 추가되는 것들을 떠나서, ExDoc가 HTML 문서를 한번에 생성한다는 것이 엄청 흥미롭게 느껴질 겁니다.
 
+**Note:** `@spec` 어노테이션은 코드의 정적 분석에 사용됩니다. 여기에 대해 배우려면, [사양과 타입](../advanced/typespec) 강의를 확인하세요.
```
Comment (marocchino) [inline ko/lessons/basics/documentation.md]: Note that, this link will be 404 for now. It is not translated yet.

File: ko/lessons/basics/testing.md (hunk 1 above)
Comment (marocchino) [inline ko/lessons/basics/testing.md]: fixed in 06deb13

File: ko/lessons/basics/testing.md [hunk 3]
```diff
@@ -79,6 +79,46 @@ Finished in 0.03 seconds (0.02s on load, 0.01s on tests)
 
 에러가 발생하는 상황을 테스트로 작성해야 할 때가 있습니다. 이럴 때 `assert_raise`를 사용하면 됩니다. 나중에 Plug를 다룰 수업에서 `assert_raise`를 사용하는 예제를 살펴보겠습니다.
 
+### assert_receive
+
+메시지를 주고 받는 엑터와 프로세스로 구성된 Elixir 애플리케이션에서는 메시지를 보낸 후를 테스트 하고 싶을 때가 많습니다. ExUnit이 자신의 프로세스 안에서 실행되기 때문에 다른 프로세스 처럼 메시지를 받을 수 있고 `assert_receive` 메크로로 assert할 수 있습니다.
```
Comment (devleoper) [inline ko/lessons/basics/testing.md]: Is there any reason to translate **macro** in **메크로** instead of **매크로**?

Comment (marocchino) [inline ko/lessons/basics/testing.md]: nope. just typo.

**Comment by devleoper** [comment]
Good to go! I think @Dalgona also loves this change.

**Comment by Dalgona** [comment]
Absolutely.

**Pull request merged**